SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp

```diff
@@ -40,46 +40,64 @@ FORCEINLINE UClass* ResolveClass(FString& ClassPath)
 	return Class;
 }
 
-void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
+ESchemaComponentType GetRpcType(UFunction* RemoteFunction)
 {
-	checkf(IsSupportedClass(Class), TEXT("Could not find class in schema database: %s"), *Class->GetPathName());
-
-	TSharedRef<FClassInfo> Info = ClassInfoMap.Add(Class, MakeShared<FClassInfo>());
-	Info->Class = Class;
-
-	TArray<UFunction*> RelevantClassFunctions = improbable::GetClassRPCFunctions(Class);
+	EFunctionFlags Flags = RemoteFunction->FunctionFlags;
 
-	for (UFunction* RemoteFunction : RelevantClassFunctions)
+	if (Flags & FUNC_NetMulticast)
 	{
-		ESchemaComponentType RPCType = SCHEMA_Invalid;
-		if (RemoteFunction->FunctionFlags & FUNC_NetClient)
-		{
-			RPCType = SCHEMA_ClientRPC;
-		}
-		else if (RemoteFunction->FunctionFlags & FUNC_NetServer)
+		return SCHEMA_NetMulticastRPC;
+	}
+	if (Flags & FUNC_NetCrossServer)
```

Comment:
Can we use `HasAnyFunctionFlags` for all of these ifs? Could we also make the ifs into else ifs?

Reply:
Done